components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -199,17 +212,19 @@ private static void populateHandlers(OMElement apiElement, InternalAPI api) {
         api.setHandlers(handlerList);
     }
 
-    private static InternalAPIHandler createHandler(String classFQName) {
+    private static InternalAPIHandler createHandler(String classFQName, String context, List<String> resources) {
 
         try {
-            Object obj = Class.forName(classFQName).newInstance();
+            Constructor c = Class.forName(classFQName).getConstructor(String.class);
+            Object obj = c.newInstance(context);
             if (obj instanceof InternalAPIHandler) {
+                ((InternalAPIHandler) obj).setResources(resources);
```

Comment:
Shall we assign the casted object to a variable and use it as we are again casting while returning.

Reply:
done